catalog/ui/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/forms/commands/SearchFormsManageCommandTest.java

```diff
@@ -0,0 +1,172 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.forms.commands;
+
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.core.StringContains.containsString;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.refEq;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.common.collect.ImmutableList;
+import ddf.catalog.CatalogFramework;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.operation.DeleteRequest;
+import ddf.catalog.operation.DeleteResponse;
+import ddf.catalog.operation.impl.DeleteRequestImpl;
+import ddf.catalog.source.IngestException;
+import ddf.catalog.source.SourceUnavailableException;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.Collections;
+import java.util.List;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class SearchFormsManageCommandTest {
+  @Mock CatalogFramework catalogFramework;
+
+  @Mock DeleteResponse deleteResponse;
+
+  @Mock Metacard mockMetacard1;
+
+  @Mock Metacard mockMetacard2;
+
+  private static ByteArrayOutputStream buffer;
+
+  private static PrintStream realSystemOut;
+
+  private SearchFormsManageCommand cmd;
+
+  @BeforeClass
+  public static void setupOutput() {
+    interceptSystemOut();
+  }
+
+  @Before
+  public void setup() throws SourceUnavailableException, IngestException {
+    when(mockMetacard1.getId()).thenReturn("1");
+    when(mockMetacard2.getId()).thenReturn("2");
+    when(mockMetacard1.getTitle()).thenReturn("title1");
+    when(mockMetacard2.getTitle()).thenReturn("title2");
+    when(deleteResponse.getDeletedMetacards()).thenReturn(ImmutableList.of(mockMetacard1));
+    when(catalogFramework.delete(any(DeleteRequest.class))).thenReturn(deleteResponse);
+    cmd =
+        new SearchFormsManageCommand() {
```

Comment:
I think we could create a mock `EndpointUtil` instead and inject it just like you're already injecting the catalog framework, then set up some stubs on that. It's still not a great solution but it seems a bit less bad than extending the command class and overriding a method. Then `retrieveSystemMetacards` wouldn't have to be `protected`. And for what it's worth, that is generally how we inject mocked dependencies in our existing command unit tests.